src/delegateCollateral.ts

```diff
@@ -12,10 +12,10 @@ export async function delegateCollateral({
   provider: ethers.providers.Web3Provider;
   walletAddress: string;
   CoreProxyContract: { address: string; abi: string[] };
-  accountId: ethers.BigNumber;
-  poolId: ethers.BigNumber;
-  tokenAddress: string;
-  delegateAmount: ethers.BigNumber;
+  accountId: BigNumberish;
+  poolId: BigNumberish;
+  tokenAddress: BigNumberish;
+  delegateAmount: BigNumberish;
```

Comment:
```suggestion
  accountId: ethers.BigNumberish;
  poolId: ethers.BigNumberish;
  tokenAddress: ethers.BigNumberish;
  delegateAmount: ethers.BigNumberish;
```

And in other places please

Reply:
fix